shelley/chain-and-ledger/cddl-spec/shelley.cddl

```diff
@@ -0,0 +1,136 @@
+; Shelley Types
+
+; When you have something that is always length 2, why do you need the header?
+; Would using a map here make it cheaper?
+block =
+  [ header                     : header
+  , transaction_bodies         : [* transaction_body]
+  , transaction_witness_sets   : [* transaction_witness_set]
+  ]
+
+; TODO: Replace direct inlining with unwrap operator (~) for closer correspondence to Haskell types
+; Inlining the header body here to save word, but this would mean no valid CBOR instance for header body...
+header =
+  [ prev_hash        : $hash
+  , issuer_vkey      : $vkey
+  , vrf_vkey         : $vrf_vkey
+  , slot             : uint
+  , nonce            : uint
```

Comment:
Do we not want to use a bitsize that is larger than a standard int ?

Reply:
that's an excellent point. the cddl file is not in sync with the spec for everything. this PR was focusing on the details up to the transaction level. nonces right now are actually being serialized as:
`TkListLen 1 . TkWord 0` for the neutral nonce, and `TkListLen 2 . TkWord 1 . TkBytes b` otherwise. we can definitely do something differently if something else is better, or change the cddl. but going too far past transactions in this PR is a bit of a can of worms :)